src/devices/IS31FL3730/IS31FL3730.cs

```diff
@@ -0,0 +1,202 @@
+            {
+                case MatrixLayout.Matrix8by8:
+                    configuration = (byte)(configuration | MATRIX_MODE_8X8);
+                    break;
+                case MatrixLayout.Matrix7by9:
+                    configuration = (byte)(configuration | MATRIX_MODE_7X9);
+                    break;
+                case MatrixLayout.Matrix6by10:
+                    configuration = (byte)(configuration | MATRIX_MODE_6X10);
+                    break;
+                case MatrixLayout.Matrix5by11:
+                    configuration = (byte)(configuration | MATRIX_MODE_5X11);
+                    break;
+            }
+
+            switch (_configuration.Mode)
+            {
+                case MatrixMode.Matrix1Only:
+                    configuration = (byte)(configuration | MATRIX_MODE_1ONLY);
+                    break;
+                case MatrixMode.Matrix2Only:
+                    configuration = (byte)(configuration | MATRIX_MODE_2ONLY);
+                    break;
+                case MatrixMode.Both:
+                    configuration = (byte)(configuration | MATRIX_MODE_BOTH);
+                    break;
+            }
+
+            _i2cDevice.Write(new ReadOnlySpan<byte>(new byte[] { MATRIX_COMMAND_CONFIGURATION_REGISTER, configuration }));
+        }
+
+        /// <summary>
+        /// Sets the LED Drive Strength register.
+        /// </summary>
+        public void SetDriveStrength(DriveStrength driveStrength)
+        {
+            _configuration.DriveStrength = driveStrength;
+            SetDriveStrength();
+        }
+
+        /// <summary>
+        /// Sets the LED Drive Strength register.
+        /// </summary>
+        protected void SetDriveStrength()
+        {
+            _i2cDevice.Write(new ReadOnlySpan<byte>(new byte[] { MATRIX_COMMAND_DRIVE_STRENGTH, (byte)_configuration.DriveStrength }));
+        }
+
+        /// <summary>
+        /// Set the matrix output.
+        /// </summary>
+        /// /// <param name="matrix">Which matrix to set.</param>
+        /// <param name="display">Values to load into the matrix.</param>
+        public void SetMatrix(MatrixMode matrix, byte[] display)
+        {
+            switch (matrix)
+            {
+                case MatrixMode.Matrix1Only:
+                    _matrix1 = display;
+                    break;
+                case MatrixMode.Matrix2Only:
+                    _matrix2 = display;
+                    break;
+                case MatrixMode.Both:
+                    _matrix1 = display;
+                    _matrix2 = display;
+                    break;
+            }
+
+            UpdateDisplay();
+        }
+
+        /// <summary>
+        /// Show the internally stored matrix data on the LED displays.
+        /// </summary>
+        protected void UpdateDisplay()
+        {
+            _i2cDevice.Write(new ReadOnlySpan<byte>((new byte[] { MATRIX_COMMAND_MATRIX1 }).Concat(_matrix1).ToArray()));
```

Comment:
You may consider doing something like this rather than allocating all the times list and arrays, it's clearly not necessary and complexify everything:

```csharp
// 2 for the command and 11 for the matrix
Span<byte> toSend = stackalloc byte [13];
MATRIX_COMMAND_MATRIX1 .CopyTo(toSend.Slice(0,2));
_matrix1.CopyTo(toSend.Slice(2));
_i2cDevice.Write(toSend);
MATRIX_COMMAND_UPDATE.CopyTo(toSend.Slice(0,2))
_i2cDevice.Write(tSend(0,2));
// And continue
```

Also, consider renaming the byte array in uppercase to something that respects the naming conventions.